### PR TITLE
Only inject netrc to trusted clone plugins

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

**Comment by 6543** [comment]
asap this is implemented, I think we can make our pipeline non-gated again :)

**Comment by 6543** [comment]
Uh nice i wanted it to head into that direction - just overlooked it atm

File: shared/utils/strings.go [hunk 1]
```diff
@@ -56,3 +56,13 @@ func sliceToCountMap(list []string) map[string]int {
 	}
 	return m
 }
+
+func SliceContainsString(slice []string, s string) bool {
```
Comment (6543) [inline shared/utils/strings.go:60]: make a generic func ?

File: pipeline/frontend/yaml/compiler/compiler.go [hunk 2]
```diff
@@ -185,9 +187,14 @@ func (c *Compiler) Compile(conf *yaml.Config) (*backend.Config, error) {
 
 			name := fmt.Sprintf("%s_clone_%d", c.prefix, i)
 			step := c.createProcess(name, container, defaultCloneName)
-			for k, v := range c.cloneEnv {
-				step.Environment[k] = v
+
+			// only inject netrc to trusted plugins or if it's a trusted repo
+			if c.trustedPipeline || (container.IsPlugin() && (container.Image == cloneImage || container.IsTrusted())) {
```
Comment (6543) [inline pipeline/frontend/yaml/compiler/compiler.go:192]:
```suggestion
			// only inject netrc if it's a trusted repo or a trusted plugin
			if c.trustedPipeline || (container.IsPlugin() && container.IsTrusted()) {
```

File: pipeline/frontend/yaml/compiler/compiler.go [hunk 3]
```diff
@@ -185,9 +187,14 @@ func (c *Compiler) Compile(conf *yaml.Config) (*backend.Config, error) {
...
 			step := c.createProcess(name, container, defaultCloneName)
-			for k, v := range c.cloneEnv {
-				step.Environment[k] = v
+
+			// only inject netrc if it's a trusted repo or a trusted plugin
+			if c.trustedPipeline || (container.IsPlugin() && container.IsTrusted()) {
+				for k, v := range c.cloneEnv {
+					step.Environment[k] = v
+				}
 			}
```
Comment (anbraten) [inline pipeline/frontend/yaml/compiler/compiler.go:196]: How is this related to `WithNetrc` from the compiler? 🤔

**Comment by 6543** [comment]
not a bad idea...

File: pipeline/frontend/yaml/container.go [hunk 4]
```diff
@@ -114,3 +116,7 @@ func (c *Containers) UnmarshalYAML(value *yaml.Node) error {
 func (c *Container) IsPlugin() bool {
 	return len(c.Commands) == 0
 }
+
+func (c *Container) IsTrusted() bool {
```
Comment (6543) [inline pipeline/frontend/yaml/container.go:120]: can we call it HasTrustedCloneImage() or so?

File: server/api/repo.go [hunk 5]
```diff
@@ -153,6 +153,9 @@ func PatchRepo(c *gin.Context) {
 	if in.CancelPreviousPipelineEvents != nil {
 		repo.CancelPreviousPipelineEvents = *in.CancelPreviousPipelineEvents
 	}
+	if in.NetrcOnlyTrusted != nil {
```
Comment (6543) [inline server/api/repo.go:156]: can we make it true by default? -> PostRepo()

**Review by jolheiser — APPROVED** [review]
lgtm

**Comment by 6543** [comment]
haha we hit the rate limit :laughing:
